R/module_teal_data.R

```diff
@@ -65,6 +65,19 @@ srv_teal_data <- function(id,
     data_out <- data_module(id = "data")
     data_handled <- reactive(tryCatch(data_out(), error = function(e) e))
     observeEvent(data_handled(), {
+      if (inherits(data_out, "reactive.event")) {
+        # This warning message partially detects when `eventReactive` is used in `data_module`.
+        logger::log_warn(
```

Comment:
Please remember we are using `teal.logger::register_handlers` and `warnings` are automatically converted into `logger::log_warn`.
```suggestion
        warning(
```

Reply:
I think this is in the wrong place. In the constructor of `teal_transform_module` we check if the returned object is a `reactive`. I think this warning can be moved there. Besides a warning, IMO it should be written somewhere in the `teal_transform_module` documentation - for example in `@param server ...` 

https://github.com/insightsengineering/teal/blob/d82ce144da212d763971e6dc2bb6e5094b6d2b5b/R/teal_data_module.R#L152

Reply:
Moved warning to constructor and added documentation 69bf5964d